Source/Diagnostics/SliceDiagnostic.cpp

```diff
@@ -0,0 +1,411 @@
+            slice_realbox.setLo( idim, real_box.lo(idim));
+            amrex::Print() << " slice lo is out of bounds. Modified it in dimension " << idim << " to be aligned with the domain box\n";
+        }      
+        
+        // Modify hi if input in out od bounds //
+        if ( slice_realbox.hi(idim) > real_box.hi(idim) ) {        
+            slice_realbox.setHi( idim, real_box.hi(idim));
+            amrex::Print() << " slice hi is out of bounds. Modified it in dimension " << idim << " to be aligned with the domain box\n";
+        }
+   
+        // Factor to ensure index values computation depending on index type //
+        double fac = ( 1.0 - SliceType[idim] ) * dom_geom[0].CellSize(idim) * 0.5;
+ 
+        // if dimension is reduced to one cell length //
+        if ( slice_realbox.hi(idim) - slice_realbox.lo(idim) <= 0)
+        {
+            slice_cc_nd_box.setLo( idim, slice_realbox.lo(idim) );
+            slice_cc_nd_box.setHi( idim, slice_realbox.hi(idim) );
+
+            if ( slice_cr_ratio[idim] > 1) cr_ratio[idim] = 1;
+    
+            // check for interpolation -- compute index lo with floor and ceil
+            if ( slice_cc_nd_box.lo(idim) - real_box.lo(idim) >= fac ) {            
+                slice_lo[idim] = floor( ( (slice_cc_nd_box.lo(idim) - (real_box.lo(idim)                                          + fac ) ) / dom_geom[0].CellSize(idim)) + fac * 1E-10);
+                slice_lo2[idim] = ceil( ( (slice_cc_nd_box.lo(idim) - (real_box.lo(idim)                                          + fac) ) / dom_geom[0].CellSize(idim)) - fac * 1E-10 );    
+            }            
+            else {            
+                slice_lo[idim] =  round((slice_cc_nd_box.lo(idim) - (real_box.lo(idim) ) )                                    / dom_geom[0].CellSize(idim));
+                slice_lo2[idim] =  ceil((slice_cc_nd_box.lo(idim) - (real_box.lo(idim) ) )                                    / dom_geom[0].CellSize(idim) );
+            }
+ 
+            // flag for interpolation -- if reduced dimension location does not align with data point 
+            if ( slice_lo[idim] == slice_lo2[idim]) {            
+               if ( slice_cc_nd_box.lo(idim) - real_box.lo(idim) < fac ) {               
+                  interp_lo[idim] = 1;
+               }
+            }
+            else {            
+               interp_lo[idim] = 1;
+            }
+
+            // ncells = 1 if dimension is reduced //
+            slice_hi[idim] = slice_lo[idim] + 1;
+        }
+        else
+        {
+            // moving realbox.lo and reabox.hi to nearest coarsenable grid point //
+            int index_lo = floor( ( (slice_realbox.lo(idim) - (real_box.lo(idim)) )                                         / dom_geom[0].CellSize(idim) ) +fac * 1E-10);
+            int index_hi = ceil( ( (slice_realbox.hi(idim) - (real_box.lo(idim)) )                                         / dom_geom[0].CellSize(idim) ) - fac * 1E-10);
```

Comment:
In this line, the global index of the lo and hi of the slice are computed such that it shares the same index space as the domain.lo and hi
It should be slice_realbox.hi(idim) - real_box.lo(idim), since the index space is same as that of the domain and real_box.lo is subtracted to shift the indices accordingly, i.e., if slice_realbox.hi()==real_box.lo(idim), then its index will be 0!
If we subtract real_box.hi, we will end up with a negative index value